JS/cart.js

```diff
@@ -147,99 +162,113 @@ function createCart2(article) {
      * We update the SessionStorage by deleting an entry .
      * We also update the component that show the number of article in the cart.
     */
-    del_text_div.onclick = function () {
+    del_text_div.onclick = function () {/*
         if (confirm('Etes vous sûr de vouloir retirer cet article du panier? ')) {
             cart_div.removeChild(grid_article_in_cart);
             cart_div.removeChild(line);            
-            var old_data_saved = JSON.parse(sessionStorage.getItem('articleToCart2'));
-            var tab = [];
-            for (i = 0; i < Object.keys(old_data_saved).length; i++) {
-                if (!(old_data_saved[i].id == article.id)) {
-                    tab.push(old_data_saved[i]);
-                }
-            }
-            sessionStorage.setItem('articleToCart2', JSON.stringify(tab));
-            document.getElementById("nb_object_in_cart").textContent = "Votre panier (" + (Object.keys(old_data_saved).length - 1) + ")";
-            nb_article_in_cart.textContent = Object.keys(old_data_saved).length - 1;            
+            let old_data_saved = JSON.parse(sessionStorage.getItem('articleToCart2'));
+            old_data_saved = old_data_saved.filter(del => del.id != article.id);
+            sessionStorage.setItem('articleToCart2', JSON.stringify(old_data_saved));
+            document.getElementById("nb_object_in_cart").textContent = "Votre panier (" + old_data_saved.length + ")";
+            nb_article_in_cart.textContent = old_data_saved.length;            
+        }*/
+        document.getElementById("overlay_alert").style.display = "block";
+        document.getElementById("popupAlert").style.display = "block";
+        id_article_deleted = article.id;
+        grid_article_in_cart_deleted = grid_article_in_cart;
+        line_deleted = line;
+    }
+
+    document.getElementById("overlay_alert_btn_yes").onclick = function () {
```

Comment:
I know addEventListener but I don't understand neither **why** nor **how** you want me to use it here.

Reply:
How 
`document.getElementById("overlay_alert_btn_yes").addEventListener('click', function() { /* do stuff here*/ });`

Why : 
- allow to have multiple handlers for the same => future proof
- ability to make the event passive => better perf wise
 
I really recommend you to read the doc I put in my first comment if you haven't already done because it contains a whole lot of cool information :)

Reply:
Done for all the oninput or onclick in the cart.js